core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/SPARQLParser.java

```diff
@@ -57,6 +59,27 @@
 
 @SuppressWarnings("deprecation")
 public class SPARQLParser implements QueryParser {
+	private final Map<String, String> customPrefixes;
+
+	/**
+	 * Create a new SPARQLParser.
+	 * 
+	 * @param customPrefixes the default namespaces to apply to this parser. null for no prefixes
+	 */
+	public SPARQLParser(Set<Namespace> customPrefixes) {
+		if (customPrefixes == null || customPrefixes.isEmpty()) {
+			this.customPrefixes = Collections.emptyMap();
+		} else {
+			this.customPrefixes = Namespaces.asMap(customPrefixes);
+		}
+	}
+
+	/**
+	 * Create a new SPARQLParser without any default prefixes.
+	 */
+	public SPARQLParser() {
+		this(null);
+	}
```

Comment:
I feel that since we have two constructors then we shouldn't allow null. Could you add a null check with `Objects.requireNonNull` in the first constructor and pass in an empty set from the second constructor?

Reply:
done